apps/dotcdn/src/app/dotcdn.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@angular/core';
+import { Observable, ReplaySubject } from 'rxjs';
+import { CoreWebService, ResponseView, SiteService } from '@dotcms/dotcms-js';
+import { pluck, mergeMap } from 'rxjs/operators';
+import * as moment from 'moment';
+import { DotCDNStats, PurgeUrlOptions } from './app.interface';
+@Injectable({
+    providedIn: 'root'
+})
+export class DotCDNService {
+    currentSite$: ReplaySubject<string> = new ReplaySubject();
+
+    constructor(private coreWebService: CoreWebService, private siteService: SiteService) {
+        this.siteService
+            .getCurrentSite()
+            .pipe(pluck('identifier'))
+            .subscribe((hostId: string) => {
+                this.currentSite$.next(hostId);
+            });
+    }
+
+    /**
+     * Request stats via Core Web Service
+     *
+     * @param {string} period
+     * @return {*}  {*}
+     * @memberof DotCDNService
+     */
+    requestStats(period: string): Observable<ResponseView<DotCDNStats>> {
+        return this.currentSite$.pipe(
+            mergeMap((hostId: string) => {
+                const dateTo = moment().format('YYYY-MM-DD');
+                const dateFrom = moment().subtract(period, 'd').format('YYYY-MM-DD');
+                return this.coreWebService.requestView({
+                    url: `/api/v1/dotcdn/stats?hostId=${hostId}&dateFrom=${dateFrom}&dateTo=${dateTo}`
+                });
+            })
+        );
```

Comment:
I know we need to fix this for site change, but @alfredo-dotcms did a change in the site service recently, is in master, please bring master to this branch so when we work on the fix here we have that change because it could affect our solution.